### ISSUE TITLE
Dependabot shows an alert which it says is no longer vulnerable. 

### ISSUE THREAD

**Issue by root-lib** [dependabot/dependabot-core#4222]
There are couple of alerts eg: [No security update is needed as browserslist is no longer vulnerable](https://github.com/advisories/GHSA-w8qv-6jwh-64r5 ) where the patched version is 4.16.5 however the app updated the dependency to 4.17.0 , so technically the issue should go away. 

So what we are observing is that,  alert is still present , however on details page it says a message "alert is no longer vulnerable" . 
<img width="959" alt="Screenshot 2021-09-14 at 11 08 03 AM" src="https://user-images.githubusercontent.com/84492897/133201259-5fc3fd4f-f592-4836-9549-d89b3b4b2e8a.png">

Looking into vulnerability itself, it says nothing about versions greater than 4.16.5 , so maybe thats why alert persist(this is my guess) . I believe this is a bug as the alert should be removed .

package manager-> npm 
package -> browserlist
CVE --> CVE-2021-23364 
language- > JavaScript 
package lock file -> ui/yarn.lock

dependabot.yml :
version: 2
updates:
- package-ecosystem: npm
  directory: "/"
  target-branch: "dependabot-updates"
  schedule:
    interval: monthly
- package-ecosystem: npm
  directory: "/ui"
  target-branch: "dependabot-updates"
  schedule:
    interval: monthly

**Comment by lucasgonze** [comment]
I am seeing this alert on a dependency chain that appears to be moot because the vulnerable version is superceded by a patched version later in the same yarn.lock file. 

````yarn.lock:

ansi-regex@^2.0.0:
  version "2.1.1"
  resolved "https://registry.yarnpkg.com/ansi-regex/-/ansi-regex-2.1.1.tgz#c3b33ab5ee360d86e0e628f0468ae7ef27d654df"
  integrity sha1-w7M6te42DYbg5ijwRorn7yfWVN8=

ansi-regex@^5.0.0:
  version "5.0.0"
  resolved "https://registry.yarnpkg.com/ansi-regex/-/ansi-regex-5.0.0.tgz#388539f55179bf39339c81af30a654d69f87cb75"
  integrity sha512-bY6fj56OUQ0hU1KjFNDQuJFezqKdrAyFdIevADiqrWHwSlbmBNMHp5ak2f40Pm8JTFyM2mqxkG6ngkHO11f/lg==

has-ansi@^2.0.0:
  version "2.0.0"
  resolved "https://registry.yarnpkg.com/has-ansi/-/has-ansi-2.0.0.tgz#34f5049ce1ecdf2b0649af3ef24e45ed35416d91"
  integrity sha1-NPUEnOHs3ysGSa8+8k5F7TVBbZE=
  dependencies:
    ansi-regex "^2.0.0"

strip-ansi@6.0.0:
  version "6.0.0"
  resolved "https://registry.yarnpkg.com/strip-ansi/-/strip-ansi-6.0.0.tgz#0b1571dd7669ccd4f3e06e14ef1eed26225ae532"
  integrity sha512-AuvKTrTfQNYNIctbR1K/YGTR1756GycPsg7b9bdV9Duqur4gv6aKqHXah67Z8ImS7WEz5QVcOtlfW2rZEugt6w==
  dependencies:
    ansi-regex "^5.0.0"

strip-ansi@^3.0.0:
  version "3.0.1"
  resolved "https://registry.yarnpkg.com/strip-ansi/-/strip-ansi-3.0.1.tgz#6a385fb8853d952d5ff05d0e8aaf94278dc63dcf"
  integrity sha1-ajhfuIU9lS1f8F0Oiq+UJ43GPc8=
  dependencies:
    ansi-regex "^2.0.0"
````

**Comment by soumyamahunt** [comment]
Seeing the same issue as well, even after I merged dependabot's security fix, I am seeing this error instead of dependabot just closing this security alert.

**Comment by jeffwidman** [comment]
👋 Hey there! Sorry for the glacially slow response... we've been a bit buried the past few years, and only just now starting to dig through some of our backlog.

This issue is pretty old, and we've pushed a lot of bugfixes since then... is this still a valid / reproducible issue?

Also, let's limit the discussion in this issue to `browserlist`, as the root cause for the problem can vary depending on the package it's referring to...

**Comment by G-Rath** [comment]
@jeffwidman we have this a lot on our repos, though not specifically with `browserlist` - whats the best way to report these to you if you want them individually?

**Comment by halo** [comment]
Hi, no need to say sorry :) In 2021 I had outdated warnings for ruby gems. Those have since disappeared, so I consider that part fixed.

**Comment by jeffwidman** [comment]
@G-Rath Good question... I'm not sure, but I've asked internally and will circle back.

**Comment by G-Rath** [comment]
@jeffwidman cheers, I've got at least three to report :)

**Comment by lucasgonze** [comment]
> This issue is pretty old, and we've pushed a lot of bugfixes since then... is this still a valid / reproducible issue?

This particular item is gone now, for the moment, I assume.

**Comment by jeffwidman** [comment]
So I learned a little more from teammates about `npm` (an ecosystem that I've never worked with directly myself).  Here's my current understanding:
1. The alerts are triggered by manifest parsing code that is a separate code path from `dependabot-core`'s parsing code. So if the alert manifest parsing code thinks the repo is still vulnerable, the alert will persist. 
2. `npm` allows multiple versions of a dependency in the dep tree... and these _are_ used in the code at runtime... the newest is _not_ superseding the old one. So if those exist, you're still vulnerable.
3. `dependabot-core` has a known bug where it only updates the lowest version of a dependency... so Dependabot may try to create a PR, then report that `npm` is no longer vulnerable, when in fact a later version of the dependency _is_ vulnerable and _is_ still in the tree. That's tracked in #5741 
4. The alerts are generated based on information in the [GitHub advisory database](https://github.com/advisories). So if that database lists all versions > 0.16.5 are vulnerable, but the package just published v0.17.0 which fixes the issue, then the alert will not disappear until the advisory database is updated. Thankfully, _you_ can [submit a PR to the advisory DB to fix an incorrect version specifier](https://docs.github.com/en/code-security/dependabot/dependabot-alerts/editing-security-advisories-in-the-github-advisory-database).

So I suspect that the majority of these cases are symptoms of #6741.

As far as reporting additional cases, first take a look at your manifests to see if it matches the behavior of #6741. If it's something different, then file a new issue. 

If at all possible, please include a link to a public repo that can reproduce the problem. That should generally be possible, but if for some reason it's not you can file a support ticket instead with a link to your private repo.

And to be clear, we don't mind more issues, as long as you've done your homework to ensure they're valid.

In the meantime, I'm going to close this issue as the original `browserlist` issue is resolved and if we leave it open it'll continue to be a "catch-all" ticket... Unfortunately catch-call issues aren't really helpful because while the symptoms may be similar, the root cause may be entirely different.
